crates/storage/db-api/src/database.rs

```diff
@@ -5,13 +5,27 @@ use crate::{
 };
 use std::{fmt::Debug, sync::Arc};
 
+/// A trait for database configuration, used as the main entrypoint for opening or creating a
+/// database.
+pub trait DatabaseConfig {
```

Comment:
imo this makes a lot of sense, this way we could also support initialization setups and it is no longer tied to mdbx